internal/batches/repozip/fetcher.go

```diff
@@ -336,7 +337,10 @@ func fetchRepositoryFile(ctx context.Context, client HTTPClient, repo RepoRevisi
 		return false, fmt.Errorf("unable to fetch archive (HTTP %d from %s)", resp.StatusCode, req.URL.String())
 	}
 
-	f, err := os.Create(dest)
+	tmpFileDir := fmt.Sprintf("%s.%d.tmp", dest, rand.Int())
+	// Ensure the file doesn't exist.
+	_ = os.Remove(tmpFileDir)
+	f, err := os.Create(tmpFileDir)
```

Comment:
We should use [`os.CreateTemp`](https://pkg.go.dev/os#CreateTemp) here — it does essentially the same thing, but in a slightly safer way. (We also need to ensure it creates the file in the same directory that it will end up in, as I think you already know, since rename operations usually can't cross filesystem boundaries.)

Reply:
how about 8c04c68a196fe7b049cb6ff08406ed7378b92bff

Reply:
I think we'll need a `defer func(path string) { _ = os.Remove(path) }(f.Name())` after the `CreateTemp` call, since there's a possible return path before the `Rename`, but otherwise LGTM.

Reply:
Makes sense. I also made sure we close the file before we rename it, although on macOS at least that doesn't seem to make a difference.

Reply:
Yeah, I think that would only matter on Windows, and I'm not even sure there. No harm in being safe, though!